sound/soc/sof/sof-audio.h

```diff
@@ -88,8 +90,16 @@ struct snd_sof_widget {
 	int core;
 	int id;
 
+	/*
+	 * Flag indicating if the widget should be set up dynamically when a PCM is opened.
+	 * If not set, then the widget is set up at the time of topology loading and retained
+	 * until the DSP enters D3. It will need to set up again when resuming from D3.
+	 */
+	int dynamic_pipeline_widget;
+
 	struct snd_soc_dapm_widget *widget;
 	struct list_head list;	/* list in sdev widget list */
+	struct snd_sof_widget *pipe_widget;
```

Comment:
Would be good to explain in the commit message why this is added. The usage might come later, but to review individual steps, a brief explanation would help.

Reply:
@ranj063  Ack, this is not covered in commit message. Resolving conversation.